### PR TITLE
Add webhook to prevent duplicate KataConfig instances

### PR DESCRIPTION
**Pull request by pmores**
A KataConfig instance is a representation of Kata installation on cluster worker nodes therefore it'd be hard at this point to assign a meaningful semantics to multiple KataConfig instances in a single cluster.  Yet it's been possible so far to create multple Kata instances.

This PR fixes the issue by adding a validating webhook for KataConfig resources that prevents creation of additional instances if one already exists in a cluster.

The fix can be verified by attempting to create a duplicate KataConfig instance.  The expected result is an error message saying that the operation failed due to being denied by the webhook. 

**- Description for the changelog**
Prevent creation of duplicate KataConfig instances.

### CODE REVIEW

**Review by jensfr — APPROVED** [review]
Hi Pavel, it looks good to me. Thank you!

File: api/v1/zz_generated.deepcopy.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ package v1
 
 import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	runtime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime"
```
[the remark this answers is not in the document]
Comment (pmores) [reply api/v1/zz_generated.deepcopy.go:25]: Yes, semantically definitely unrelated (in fact, if I understand Golang correctly this change is semantically void :-)).

On the other hand, this edit was made automatically by operator-sdk or the project build process (not sure which) and may have been caused by project version upgrade so I figured I'd leave it in, out of piety towards operator-sdk, and also understanding it'd probably have to go in at some point anyway. ;-)

**Review by bpradipt — APPROVED** [review]
/lgtm

**Comment by jensfr** [comment]
@pmores There's some code in controllers/openshift_controller.go isOldestCR() that we used to prevent two kataconfigs being created before we had a webhook. I think we can (have to?) remove at least part of this code now. What do you think?

**Comment by pmores** [comment]
@jensfr I agree, this should be obsolete now although I'd prefer to remove it in a separate PR.  The obsolete code shouldn't interfere with webhook operation - the webhook runs before our reconciliation loop which means the old code guarding against duplicate KataConfigs should simply never trigger anymore.

**Comment by jensfr** [comment]
We already have a test case to test duplicate KataConfig's. So I think we can merge this now.

**Pull request merged**